tests/contrib/django/test_django_snapshots.py

```diff
@@ -156,28 +151,30 @@ def test_psycopg_query_default(client, psycopg2_patched):
 
 
 @pytest.mark.skipif(django.VERSION < (3, 0, 0), reason="ASGI not supported in django<3")
-@pytest.mark.snapshot(
+@snapshot(
     variants={
         "30": (3, 0, 0) <= django.VERSION < (3, 1, 0),
         "31": (3, 1, 0) <= django.VERSION < (3, 2, 0),
         "3x": django.VERSION >= (3, 2, 0),
     },
+    token_override="tests.contrib.django.test_django_snapshots.test_asgi_200",
 )
-def test_asgi_200(daphne_client):
-    resp = daphne_client.get("/")
+@pytest.mark.parametrize("django_asgi", ["application", "channels_application"])
+def test_asgi_200(django_asgi):
+    resp = daphne_client_request(django_asgi, "GET", "/")
     assert resp.status_code == 200
     assert resp.content == b"Hello, test app."
 
 
 @pytest.mark.skipif(django.VERSION < (3, 0, 0), reason="ASGI not supported in django<3")
-@pytest.mark.snapshot(
+@snapshot(
```

Comment:
why this change? what is the different between `pytest.mark.snapshot` and `snapshot`?

Reply:
pytest.mark.snapshot overrides the existing snapshot fixture and that fixture was used in the daphne client fixture. This is what I was trying to wrap my head around on Monday. With the snapshot annotation I can override the snapshot token and generate one snapshot for both asgi_200[application] and asgi_200[channels_application] tests, I can't do this with the snapshot fixture (I can but it would require a code change that I am not sure we want to do in this PR)